src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -100,44 +101,93 @@ def run_pipeline(
     # in the data
     else:
-        try:
-            # First attempt is storing the message directly in the
-            # invalid messages container
-            store_data(
-                container_url,
-                invalid_output_path,
-                f"{message_mappings['filename']}.{message_mappings['file_suffix']}",
-                message_mappings["bundle_type"],
-                message=message,
-            )
-        except ResourceExistsError:
-            logging.warning(
-                "Attempted to store preexisting resource: "
-                + f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
-            )
-        try:
-            # Then, try to store the conversion response information
-            store_data(
-                container_url,
-                invalid_output_path,
-                f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
-                + ".convert-resp",
-                message_mappings["bundle_type"],
-                message_json=response,
-            )
-        except ResourceExistsError:
-            logging.warning(
-                "Attempted to store preexisting resource: "
-                + f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
-                + ".convert-resp"
-            )
+        __record_error_response(
+            container_url=container_url,
+            output_path=invalid_output_path,
+            message_mappings=message_mappings,
+            transaction_type="convert",
+            message=message,
+            response=convert_response,
+        )
+
+
+def __record_error_response(
+    container_url: str,
+    output_path: str,
+    message_mappings: dict,
+    transaction_type: str,
+    message: str,
+    response: Response,
+):
+    try:
+        # First attempt is storing the message directly in the
+        # invalid messages container
+        store_data(
+            container_url,
+            output_path,
+            f"{message_mappings['filename']}.{message_mappings['file_suffix']}",
+            message_mappings["bundle_type"],
+            message=message,
+        )
+    except ResourceExistsError:
+        logging.warning(
+            "Attempted to store preexisting resource: "
+            + f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
+        )
+    try:
+        # Then, try to store the response information
+        store_data(
+            container_url,
+            output_path,
+            f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
+            + f".{transaction_type}-resp",
+            message_mappings["bundle_type"],
+            message_json=response.json(),
+        )
+    except ResourceExistsError:
+        logging.warning(
+            "Attempted to store preexisting resource: "
+            + f"{message_mappings['filename']}.{message_mappings['file_suffix']}"
+            + f".{transaction_type}-resp"
+        )
```

Comment:
Looking at this more closely I think it makes sense to move `_record_error_response` into `phdi_building_blocks.azure`, which is where `store_data` already lives. I think this would better align with our goals around modularity. Definitely happy to discuss though.

Reply:
I think this makes sense as well.  I will make the update.